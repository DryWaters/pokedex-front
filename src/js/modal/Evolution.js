import React from 'react';
import PropTypes from 'prop-types';

import { getPokemonSprite } from '../core/PokemonAPI';
import PokemonType from '../core/PokemonType';

import '../../styles/modal/Evolution.css';

const propTypes = {
  getPokemonDetails: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  types: PropTypes.arrayOf(PropTypes.string).isRequired,
  image_path: PropTypes.string.isRequired,
};

class Evolution extends React.Component {
  render () {
    return (
      <span className="evolution-container" onClick={() => { this.props.getPokemonDetails(this.props.id) }} >
        <div className="evolution">
          <img className="evolution-sprite"
               src={getPokemonSprite(this.props.image_path)}
               alt={this.props.name}
          />
        </div>
        <h5><b>{this.props.name}</b></h5>
        <PokemonType isLarge={false} isWeakness={false} types={this.props.types} />
      </span>
    );
  }
}

Evolution.propTypes = propTypes;

export default Evolution;
