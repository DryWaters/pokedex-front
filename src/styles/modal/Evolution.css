.evolution-sprite {
  width: 100%;
  height: 100%;
}

.evolution{
  width: 150px;
  height: 150px;

  border-style: solid;
  border-width: 2px;
  border-radius: 50%;
  padding: 10px;
  margin: 3px 0px;
}
