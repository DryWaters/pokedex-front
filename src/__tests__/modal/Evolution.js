import React from 'react';
import { shallow } from 'enzyme';

import Evolution from '../../js/modal/Evolution';

import ExampleJSON from '../../../local_data/example_detail.json';

describe('Evolution Component', () => {
  let props;
  beforeEach(() => {
    props = {
      getPokemonDetails: (() => {}),
      id: ExampleJSON.evolutions[1][0].id,
      name: ExampleJSON.evolutions[1][0].name,
      types: ExampleJSON.evolutions[1][0].types,
      image_path: ExampleJSON.evolutions[1][0].image_path,
    };
  });

  it('should render Evolution component without crashing', () => {
    shallow(<Evolution {...props} />);
  });

  it('should match Evolution component snapshot', () => {
    const wrapper = shallow(<Evolution {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

});
