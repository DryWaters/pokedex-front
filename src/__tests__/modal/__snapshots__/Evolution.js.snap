// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evolution Component should match Evolution component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Evolution
    getPokemonDetails={[Function]}
    id={10}
    image_path="/sprites/pokemon/small/10.png"
    name="Caterpie"
    types={
        Array [
            "bug",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="evolution"
>
          <img
                    alt="Caterpie"
                    className="evolution-sprite"
                    src="https://pokedex-backend-server.herokuapp.com/sprites/pokemon/small/10.png"
          />
</div>,
        <h5>
          <b>
                    Caterpie
          </b>
</h5>,
        <PokemonType
          isLarge={false}
          isWeakness={false}
          types={
                    Array [
                              "bug",
                            ]
          }
/>,
      ],
      "className": "evolution-container",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Caterpie"
            className="evolution-sprite"
            src="https://pokedex-backend-server.herokuapp.com/sprites/pokemon/small/10.png"
/>,
          "className": "evolution",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Caterpie",
            "className": "evolution-sprite",
            "src": "https://pokedex-backend-server.herokuapp.com/sprites/pokemon/small/10.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <b>
            Caterpie
</b>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Caterpie",
          },
          "ref": null,
          "rendered": "Caterpie",
          "type": "b",
        },
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isLarge": false,
          "isWeakness": false,
          "types": Array [
            "bug",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="evolution"
>
            <img
                        alt="Caterpie"
                        className="evolution-sprite"
                        src="https://pokedex-backend-server.herokuapp.com/sprites/pokemon/small/10.png"
            />
</div>,
          <h5>
            <b>
                        Caterpie
            </b>
</h5>,
          <PokemonType
            isLarge={false}
            isWeakness={false}
            types={
                        Array [
                                    "bug",
                                  ]
            }
/>,
        ],
        "className": "evolution-container",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Caterpie"
              className="evolution-sprite"
              src="https://pokedex-backend-server.herokuapp.com/sprites/pokemon/small/10.png"
/>,
            "className": "evolution",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Caterpie",
              "className": "evolution-sprite",
              "src": "https://pokedex-backend-server.herokuapp.com/sprites/pokemon/small/10.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <b>
              Caterpie
</b>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Caterpie",
            },
            "ref": null,
            "rendered": "Caterpie",
            "type": "b",
          },
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isLarge": false,
            "isWeakness": false,
            "types": Array [
              "bug",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
